packages/mongodb-rag-core/src/services/conversations.ts

```diff
@@ -10,6 +10,12 @@ export type Reference = z.infer<typeof Reference>;
 export const Reference = z.object({
   url: z.string(),
   title: z.string(),
+  metadata: z
+    .object({
+      sourceName: z.string().optional(),
+      tags: z.array(z.string()).optional(),
+    })
+    .optional(),
```

Comment:
is there a way that we could also have the metadata object accept arbitrary other properties? 

if we're defining, don't want to restrict to just these 2 values